lib/types/local-date.js

```diff
@@ -26,58 +22,81 @@ var twentiethCentury = new Date('1899-12-31') - new Date('0000-01-01');
  * @param {Number} month Between 1 and 12 inclusive.
  * @param {Number} day Between 1 and the number of days in the given month of the given year.
+ *
+ * @property {Date} date The date representation if falls within a range of an ES5 data type, otherwise an invalid date.
+ * @property {Number} value The number of days since epoch.
+ *
  * @constructor
  */
 function LocalDate(year, month, day) {
   //implementation detail: internally uses a UTC based date
   if (typeof year === 'number' && typeof month === 'number' && typeof day === 'number') {
-    this.value = new Date(Date.UTC(year, month - 1, day));
-    if (year >= 0 && year < 100) {
-      //If there is a 2 digit year, Date.UTC() assumes that is the 20th century
-      //Thanks ECMAScript!
-      this.value = new Date(this.value.getTime() - twentiethCentury);
+    //Use setUTCFullYear as if there is a 2 digit year, Date.UTC() assumes
+    //that is the 20th century.  Thanks ECMAScript!
+    var date = new Date();
+    date.setUTCHours(0, 0, 0, 0);
+    date.setUTCFullYear(year, month-1, day);
+    if(isNaN(date.getTime())) {
+      throw new Error(util.format('%d-%d-%d does not form a valid ES5 date!',
+        year, month, day));
     }
+
+    this.date = date;
+    this.value = Math.floor(this.date.getTime() / millisecondsPerDay);
   }
   else if (typeof month === 'undefined' && typeof day === 'undefined') {
     if (typeof year === 'number') {
-      //in milliseconds since unix epoch
-      this.value = new Date(year);
+      //in days since epoch.
+      if(year < -2147483648 || year > 2147483647) {
+        throw new Error('You must provide a valid value for days since epoch (-2147483648 <= value <= 2147483647).')
+      }
+      this.value = year;
+      this.date = new Date(this.value * millisecondsPerDay);
     }
   }
-  if (!this.value || isNaN(this.value.getTime())) {
+  if (typeof this.value === 'undefined' || isNaN(this.value)) {
     throw new Error('You must provide a valid year, month and day');
   }
   var self = this;
+
   /**
-   * A number representing the year.
+   * A number representing the year.  May return NaN if cannot be represented as
+   * a Date.
    * @name year
    * @type Number
    * @memberof module:types~LocalDate#
    */
   /**
-   * A number between 1 and 12 inclusive representing the month.
+   * A number between 1 and 12 inclusive representing the month.  May return
+   * NaN if cannot be represented as a Date.
    * @name month
    * @type Number
    * @memberof module:types~LocalDate#
    */
   /**
    * A number between 1 and the number of days in the given month of the given year (28, 29, 30, 31).
+   * May return NaN if cannot be represented as a Date.
    * @name day
    * @type Number
    * @memberof module:types~LocalDate#
    */
   Object.defineProperties(this, {
     'year': { enumerable: true, get: function () {
-      return self.value.getUTCFullYear();
+      return isNaN(self.date.getTime()) ? NaN : self.date.getUTCFullYear();
```

Comment:
I think this is unnecessary, `self.value.getUTCFullYear()` already returns `NaN` if `getTime()` is `NaN`.